src/rgw/rgw_aio.h

```diff
@@ -34,7 +32,7 @@ struct D3nGetObjData;
 namespace rgw {
 
 struct AioResult {
-  RGWSI_RADOS::Obj obj;
+  rgw_raw_obj obj;
```

Comment:
@cbodley : In case of non rados stores/ filters, can pool and loc be left uninitialised?

Reply:
Also, can I use this PR now, to get past the segfault that I encountered?

Reply:
> @cbodley : In case of non rados stores/ filters, can pool and loc be left uninitialised?

yeah. i used `rgw_raw_obj` isntead of just `std::string` here because the rados backend needs to know the pool/loc of the write completions it gets, but other backends don't need them

> Also, can I use this PR now, to get past the segfault that I encountered?

it builds and passes s3tests, so it's worth a shot!